Broken compatibility with new release of SQLAlchemy 1.2.0

Hello,

Yesterday released new version of SQLAlchemy (1.2.0) and new release incompatible with aiopg:

```
mymodule.py:42: in fetchone
    result = await conn.execute(query)
.tox/py36-tests/lib/python3.6/site-packages/aiopg/utils.py:72: in __await__
    resp = yield from self._coro
.tox/py36-tests/lib/python3.6/site-packages/aiopg/sa/connection.py:116: in _execute
    return ResultProxy(self, cursor, self._dialect, result_map)
.tox/py36-tests/lib/python3.6/site-packages/aiopg/sa/result.py:234: in __init__
    self._metadata = ResultMetaData(self, cursor.description)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <aiopg.sa.result.ResultMetaData object at 0x7f0417f6a550>
result_proxy = <aiopg.sa.result.ResultProxy object at 0x7f0417f6a5c0>
metadata = (Column(name='column_0', type_code=23, display_size=None, internal_size=4, precision=None, scale=None, null_ok=None), ...Column(name='column_2', type_code=3802, display_size=None, internal_size=-1, precision=None, scale=None, null_ok=None))

    def __init__(self, result_proxy, metadata):
        self._processors = processors = []
    
        result_map = {}
        if result_proxy._result_map:
            result_map = {elem[0]: elem[3] for elem in
                          result_proxy._result_map}
    
        # We do not strictly need to store the processor in the key mapping,
        # though it is faster in the Python version (probably because of the
        # saved attribute lookup self._processors)
        self._keymap = keymap = {}
        self.keys = []
        dialect = result_proxy.dialect
>       typemap = dialect.dbapi_type_map
E       AttributeError: 'PGDialect_psycopg2' object has no attribute 'dbapi_type_map'
```

--- a/aiopg/sa/result.py
+++ b/aiopg/sa/result.py
@@ -100,7 +100,12 @@ class ResultMetaData(object):
         self._keymap = keymap = {}
         self.keys = []
         dialect = result_proxy.dialect
-        typemap = dialect.dbapi_type_map
+
+        # `dbapi_type_map` property removed in SQLAlchemy 1.2+.
+        # Usage of `getattr` only needed for backward compatibility with
+        # older versions of SQLAlchemy.
+        typemap = getattr(dialect, 'dbapi_type_map', {})
+
         assert dialect.case_sensitive, \
             "Doesn't support case insensitive database connection"
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -11,5 +11,5 @@ pytest-cov==2.5.1
 pytest-sugar==0.9.0
 pytest-timeout==1.2.1
 sphinxcontrib-asyncio==0.2.0
-sqlalchemy==1.1.15
+sqlalchemy==1.2.0
 psycopg2==2.6.2

